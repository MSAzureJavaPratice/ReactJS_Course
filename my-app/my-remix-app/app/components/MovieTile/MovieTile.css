/* MovieTile.css */

.movie-tile {
    width: 280px;
    border: 1px solid #ddd;
    padding: 10px;
    text-align: center;
    cursor: pointer;
    position: relative;
    background-color: #fff;
    /* White background for the tile */
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.movie-tile img {
    width: 100%;
    height: 100%;
    border-radius: 4px;
}

.movie-tile__menu {
    position: absolute;
    top: 10px;
    right: 10px;
    cursor: pointer;
    padding: 5px;
    /* Added padding for better spacing around the menu */
    background-color: black;
    /* Make the menu button circular */
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.2);
    /* Added shadow for visibility */
}

.menu-button {
    background: transparent;
    border: none;
    font-size: 1.5rem;
    cursor: pointer;
    color: white;
    /* Dark color for visibility */
}

.movie-tile__menu-dropdown {
    position: absolute;
    background: white;
    border: 1px solid #ccc;
    padding: 10px;
    /* Increased padding for better spacing */
    top: 30px;
    right: 0;
    border-radius: 4px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    display: none;
    /* Hidden by default, will show when menu is clicked */
}

.movie-tile__menu-dropdown button {
    background: none;
    border: none;
    padding: 8px 15px;
    /* Increased padding for the buttons */
    width: 100%;
    text-align: left;
    cursor: pointer;
    color: #333;
    /* Dark text for readability */
}

.movie-tile__menu-dropdown button:hover {
    background-color: #f0f0f0;
    color: red;
    /* Change color to red on hover */
}

.movie-tile__info {
    margin-top: 10px;
}

.movie-tile__title {
    display: flex;
    justify-content: center;
    margin-bottom: 8px;
}

.movie-tile__name {
    font-weight: bold;
    font-size: 1.1rem;
}

.movie-tile__year {
    font-size: 0.9rem;
    color: #888;
}

.movie-tile__genres {
    font-size: 0.85rem;
    color: #555;
    text-align: center;
    margin-top: 5px;
}


/* Hover effect for movie tile */

.movie-tile:hover {
    transform: scale(1.05);
}

.movie-tile:hover .movie-tile__menu-dropdown {
    display: block;
    /* Show the dropdown when tile is hovered */
}

.movie-tile {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    text-align: center;
    margin: 10px;
}

.movie-tile img {
    height: 200px;
    width: 150px;
    object-fit: cover;
    border-radius: 8px;
}

.movie-tile h3 {
    margin: 10px 0;
}

.movie-tile p {
    margin: 5px 0;
}