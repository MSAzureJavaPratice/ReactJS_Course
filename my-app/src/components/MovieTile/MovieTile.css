.movie-tile {
    width: 280px;
    border: 1px solid #ddd;
    padding: 10px;
    text-align: center;
    cursor: pointer;
    position: relative;
    /* This is necessary for positioning the menu */
    background-color: #fff;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.movie-tile img {
    width: 100%;
    height: 100%;
    border-radius: 4px;
}

.movie-tile__menu {
    position: absolute;
    top: 10px;
    right: 10px;
    cursor: pointer;
}

.menu-button {
    background: transparent;
    border: none;
    font-size: 1.5rem;
    cursor: pointer;
}

.movie-tile__menu-dropdown {
    position: absolute;
    background: white;
    border: 1px solid #ccc;
    padding: 5px;
    top: 30px;
    right: 0;
    border-radius: 4px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.movie-tile__menu-dropdown button {
    background: none;
    border: none;
    padding: 5px;
    width: 100%;
    text-align: left;
    cursor: pointer;
}

.movie-tile__menu-dropdown button:hover {
    background-color: #f0f0f0;
}

.movie-tile__info {
    margin-top: 10px;
}

.movie-tile__title {
    display: flex;
    justify-content: center;
    margin-bottom: 8px;
}

.movie-tile__name {
    font-weight: bold;
    font-size: 1.1rem;
}

.movie-tile__year {
    font-size: 0.9rem;
    color: #888;
}

.movie-tile__genres {
    font-size: 0.85rem;
    color: #555;
    text-align: center;
    margin-top: 5px;
}


/* Hover effect for movie tile */

.movie-tile:hover {
    transform: scale(1.05);
}

.movie-tile:hover .movie-tile__menu-dropdown {
    display: block;
}

.movie-tile {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    text-align: center;
    margin: 10px;
}

.movie-tile img {
    height: 200px;
    /* Set the desired height */
    width: 150px;
    /* Optionally, set a fixed width */
    object-fit: cover;
    /* Ensures the image covers the area without stretching */
    border-radius: 8px;
    /* Optional: adds rounded corners */
}

.movie-tile h3 {
    margin: 10px 0;
}

.movie-tile p {
    margin: 5px 0;
}