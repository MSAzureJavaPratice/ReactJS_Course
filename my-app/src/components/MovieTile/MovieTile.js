import React, { useState } from "react";
import "./MovieTile.css";

// MovieTile component
const MovieTile = ({ movie, onClick, onEdit, onDelete }) => {
  const { imageUrl, name, releaseYear, genres } = movie;
  const [showMenu, setShowMenu] = useState(false);

  const handleMenuClick = (e) => {
    e.stopPropagation(); // Prevent click from triggering parent onClick
    setShowMenu((prev) => !prev);
  };

  const handleMenuItemClick = (action) => {
    setShowMenu(false);
    console.log(`${action} clicked for movie:`, movie);
    if (action === "edit") {
      onEdit(movie);
    } else if (action === "delete") {
      onDelete(movie);
    }
  };

  return (
    <div className="movie-tile" onClick={() => onClick(movie)}>
      <img src={imageUrl} alt={name} className="movie-tile__image" />
      <div className="movie-tile__menu" onClick={handleMenuClick}>
        <button className="menu-button"> ... </button>{" "}
        {showMenu && (
          <div className="movie-tile__menu-dropdown">
            <button onClick={() => handleMenuItemClick("edit")}> Edit </button>{" "}
            <button onClick={() => handleMenuItemClick("delete")}>
              {" "}
              Delete{" "}
            </button>{" "}
          </div>
        )}{" "}
      </div>{" "}
      <div className="movie-tile__info">
        <div className="movie-tile__title"> {name} </div>{" "}
        <div className="movie-tile__year"> {releaseYear} </div>{" "}
        <div className="movie-tile__genres"> {genres.join(", ")} </div>{" "}
      </div>{" "}
    </div>
  );
};

export default MovieTile;
