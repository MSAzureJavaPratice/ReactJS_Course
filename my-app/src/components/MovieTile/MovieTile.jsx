import React, { useState } from "react";
import "./MovieTile.css";

// MovieTile component
const MovieTile = ({ movie, onClick, onEdit, onDelete }) => {
  const { poster_path, title, release_date, genres } = movie;
  const [showMenu, setShowMenu] = useState(false);

  const handleMenuClick = (e) => {
    e.stopPropagation(); // Prevent click from triggering parent onClick
    setShowMenu((prev) => !prev);
  };

  const handleMenuItemClick = (action) => {
    setShowMenu(false);
    if (action === "edit") {
      onEdit(movie);
    } else if (action === "delete") {
      onDelete(movie);
    }
  };

  return (
    <div className="movie-tile" onClick={() => onClick(movie)}>
      <img src={poster_path} alt={title} className="movie-tile__image" />
      <div className="movie-tile__menu" onClick={handleMenuClick}>
        <button className="menu-button">...</button>
        {showMenu && (
          <div className="movie-tile__menu-dropdown">
            <button onClick={() => handleMenuItemClick("edit")}>Edit</button>
            <button onClick={() => handleMenuItemClick("delete")}>Delete</button>
          </div>
        )}
      </div>
      <div className="movie-tile__info">
        <div className="movie-tile__title" data-testid="movie-title">{title}</div>
        <div className="movie-tile__year">{release_date.split("-")[0]}</div>
        <div className="movie-tile__genres">{genres.join(", ")}</div>
      </div>
    </div>
  );
};

export default MovieTile;
