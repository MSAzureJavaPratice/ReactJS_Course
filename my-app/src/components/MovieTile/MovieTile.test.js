import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import MovieTile from "./MovieTile";

describe("MovieTile Component", () => {
    const mockMovie = {
        imageUrl: "https://via.placeholder.com/150",
        name: "Inception",
        releaseYear: "2010",
        genres: ["Sci-Fi", "Thriller"],
    };
    const onClick = jest.fn();
    const onEdit = jest.fn();
    const onDelete = jest.fn();

    afterEach(() => {
        jest.clearAllMocks();
    });

    test("renders movie tile correctly", () => {
        render( <
            MovieTile movie = { mockMovie }
            onClick = { onClick }
            onEdit = { onEdit }
            onDelete = { onDelete }
            />
        );

        // Check for movie name, year, and genres
        expect(screen.getByText("Inception")).toBeInTheDocument();
        expect(screen.getByText("2010")).toBeInTheDocument();
        expect(screen.getByText("Sci-Fi, Thriller")).toBeInTheDocument();

        // Check if the image is rendered with correct alt text
        const image = screen.getByAltText("Inception");
        expect(image).toHaveAttribute("src", mockMovie.imageUrl);
    });

    test("calls onClick when clicked", () => {
        render( <
            MovieTile movie = { mockMovie }
            onClick = { onClick }
            onEdit = { onEdit }
            onDelete = { onDelete }
            />
        );

        fireEvent.click(screen.getByText("Inception"));
        expect(onClick).toHaveBeenCalledWith(mockMovie);
    });

    test("shows and hides menu when clicking dots", () => {
        render( <
            MovieTile movie = { mockMovie }
            onClick = { onClick }
            onEdit = { onEdit }
            onDelete = { onDelete }
            />
        );

        const menuButton = screen.getByText("...");
        fireEvent.click(menuButton);

        // Verify menu items are displayed
        expect(screen.getByText("Edit")).toBeInTheDocument();
        expect(screen.getByText("Delete")).toBeInTheDocument();

        // Click the dots again to hide the menu
        fireEvent.click(menuButton);
        expect(screen.queryByText("Edit")).not.toBeInTheDocument();
        expect(screen.queryByText("Delete")).not.toBeInTheDocument();
    });
});