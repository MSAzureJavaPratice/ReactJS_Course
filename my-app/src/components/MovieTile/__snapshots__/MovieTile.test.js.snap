// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-tile"
  >
    <img
      alt="Inception"
      class="movie-tile__image"
      src="https://via.placeholder.com/150"
    />
    <div
      class="movie-tile__menu"
    >
      <button
        class="menu-button"
      >
         ... 
      </button>
        
    </div>
     
    <div
      class="movie-tile__info"
    >
      <div
        class="movie-tile__title"
      >
         Inception 
      </div>
       
      <div
        class="movie-tile__year"
      >
         2010 
      </div>
       
      <div
        class="movie-tile__genres"
      >
         Sci-Fi, Thriller 
      </div>
       
    </div>
     
  </div>
</DocumentFragment>
`;
